centreon/src/Core/Resources/Infrastructure/Repository/DbReadResourceRepository.php

```diff
@@ -72,7 +72,8 @@ class DbReadResourceRepository extends AbstractRepositoryDRB implements ReadReso
         'alias' => 'resources.alias',
         'fqdn' => 'resources.address',
         'type' => 'resources.type',
-        'h.name' => 'parent_resource.name',
+        // We know this trick breaks the EQUAL, but this is theoretically ONLY used as REGEX from the Resource status
+        'h.name' => 'CONCAT(resources.parent_name, resources.name)',
```

Comment:
Indeed, your proposal breaks the EQUAL operator.
```suggestion
        'h.name' => 'CASE type WHEN 1 THEN `resources.name` ELSE `resources.parent_name` END',
```